lang/src/vec.rs

```diff
@@ -21,17 +22,41 @@ impl<T: ToAccountMetas> ToAccountMetas for Vec<T> {
 }
 
 impl<'info, T: Accounts<'info>> Accounts<'info> for Vec<T> {
+    /// try_accounts for Vec consumes the rest of the accounts, thus a vector should always be at the end of the struct
+    // TODO:!!!! (maybe have ix_data give it)
     fn try_accounts(
         program_id: &Pubkey,
         accounts: &mut &[AccountInfo<'info>],
         ix_data: &[u8],
     ) -> Result<Self, ProgramError> {
         let mut vec: Vec<T> = Vec::new();
-        T::try_accounts(program_id, accounts, ix_data).map(|item| vec.push(item))?;
+        while accounts.len() != 0 {
+            // let data: &[u8] = &.try_borrow_data()?;
+            // let mut disc_bytes = [0u8; 8];
+            // disc_bytes.copy_from_slice(&data[..8]);
+            T::try_accounts(program_id, accounts, ix_data).map(|item| vec.push(item))?;
+        }
+
         Ok(vec)
     }
 }
 
+impl<'info, T: AccountsExit<'info>> AccountsExit<'info> for Vec<T> {
+    fn exit(&self, program_id: &Pubkey) -> ProgramResult {
+        let res: Result<Vec<_>, ProgramError> = self
+            .iter()
+            .map(|account| account.exit(program_id))
+            .collect();
+        res.map(|_| ())
```

Comment:
This code ignores any error for underlaying `exit` call. Better to change to usual for loop, like:
```rust
for account in self {
  account.exit(program_id)?;
}
Ok(())
```
or use [Iterator::try_for_each](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.try_for_each):
```rust
self.iter().try_for_each(|account| account.exit(program_id))
```

Reply:
Awesome, and fixed!